### PR TITLE
Dynamic opcode

### PR DESCRIPTION
**Pull request by AyAggarwal**
SDK Updates for dynamic opcode accounting, backward jumps, loops, callsub, retsub

### CODE REVIEW

File: tests/8.LogicSig.js [hunk 1]
```diff
@@ -323,10 +293,25 @@ describe('Logic validation', () => {
 
       // 800x keccak256 more is to costly
```
Comment (jdtzmn) [inline tests/8.LogicSig.js:294]: nit: too

**Review by jdtzmn — CHANGES_REQUESTED** [review]

File: tests/8.LogicSig.js (hunk 1 above)
Comment (AyAggarwal) [reply tests/8.LogicSig.js]: done

File: tests/8.LogicSig.js [hunk 2]
```diff
@@ -321,12 +321,29 @@ describe('Logic validation', () => {
...
-        new Error('program too costly to run')
-      );
+      //  old versions
+      const oldVersions = [0x1, 0x2, 0x3];
+      let v;
+      for (v of oldVersions) {
+        program[0] = v;
+        assert.throws(
+          () => logic.checkProgram(program),
+          new Error('program too costly to run')
```
Comment (shiqizng) [inline tests/8.LogicSig.js:333]: shouldn't the error message have been updated to 'program too costly for Teal version < 4. consider using v4.'?

File: tests/8.LogicSig.js [hunk 3]
```diff
@@ -321,12 +321,29 @@ describe('Logic validation', () => {
...
+        );
+      }
+      //  new versions
+      const newVersions = [0x4];
+      for (v of newVersions) {
+        program[0] = v;
+        assert.throws(
+          () => logic.checkProgram(program),
+          new Error(
+            'program too costly for Teal version < 4. consider using v4.'
```
Comment (shiqizng) [inline tests/8.LogicSig.js:343]: I believe this test case should pass without error for version 4 and later since there's no more static cost calculation.

**Review by shiqizng — CHANGES_REQUESTED** [review]

File: tests/8.LogicSig.js (hunk 3 above)
Comment (AyAggarwal) [reply tests/8.LogicSig.js:343]: ah yes let me change this

**Comment by shiqizng** [comment]
Can you please also add the new unit tests for these op codes,  callsub, retsub, loop? like what you had in java sdk.

**Review by shiqizng — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]
Looks good to me

**Review by jdtzmn — APPROVED** [review]

**Pull request merged**